go/teams/transactions.go

```diff
@@ -20,24 +21,27 @@ import (
 // they are for stale versions of the addees.
 // Not threadsafe.
 type AddMemberTx struct {
-	team     *Team
-	payloads []interface{} // *SCTeamInvites or *keybase1.TeamChangeReq
-
-	// We need two separate payloads for social invites and keybase
-	// invites because they cannot be mixed in one link.
-	isForSocialInvites map[int]bool
+	team *Team
+	// payload is of either type:
+	// *txTeamInvitesKeybase, *txTeamInvitesSocial,
+	// or *keybase1.TeamChangeReq.
+	payloads []interface{}
 
 	completedInvites map[keybase1.TeamInviteID]bool
+	uidSeqno         map[keybase1.UID]int
 
 	// Override whether the team key is rotated.
 	SkipKeyRotation *bool
 }
 
+type txTeamInvitesKeybase struct{ Val SCTeamInvites }
+type txTeamInvitesSocial struct{ Val SCTeamInvites }
```

Comment:
Lots of freaky reflection going on now. I know go is lacking in this department. What do you think about one of these alternatives?

```
// Exactly one of these must be filled.
type txEntry struct {
  Cryptomembers *keybase1.TeamChangeReq
  InvitesKeybase *SCTeamInvites
  InvitesSocial *SCTeamInvites
}
```

```
type txEntryType string
const txEntryType_Cryptomembers txEntryType = "cryptomembers" // -> keybase1.TeamChangeReq
const txEntryType_InvitesKeybase txEntryType = "invites_keybase" // -> SCTeamInvites
const txEntryType_InvitesSocial txEntryType = "invites_social" // -> SCTeamInvites

type txEntry struct {
  Kind string
  Val interface{}
}
```

Reply:
That could work, thanks for thinking about it! I'm thinking about going with option 2). The helper functions that do the real type casting are still needed, and with this kind of tagging we can avoid reflection and still keep payload finding function not duplicated.

To be honest I was hesitating before using `"reflect"` but I noticed `lru/lru.go` uses it, so I decided to go for it, and only used basic features that lru uses.